stores/playground.ts

```diff
@@ -2,18 +2,38 @@ import type { Raw } from 'vue'
 import type { WebContainer, WebContainerProcess } from '@webcontainer/api'
 import type { VirtualFile } from '../structures/VirtualFile'
 
-export const PlaygroundStatusOrder = [
+export const PlaygroundStageOrder = [
   'init',
   'mount',
   'install',
   'start',
   'ready',
 ] as const
 
-export type PlaygroundStatus = typeof PlaygroundStatusOrder[number] | 'error'
+export const PlaygroundStatusOrder = [
+  'fulfilled',
+  'pending',
+  'rejected',
+  'waiting'
+] as const
+
+export type PlaygroundStage = typeof PlaygroundStageOrder[number]
+export type PlaygroundStatus = typeof PlaygroundStatusOrder[number]
+
+type PlaygroundStageStatus = {
+  [key in PlaygroundStage]: PlaygroundStatus
+}
+
+
 
 export const usePlaygroundStore = defineStore('playground', () => {
-  const status = ref<PlaygroundStatus>('init')
+  const stageStatusMap = ref<PlaygroundStageStatus>({
+    'init': 'fulfilled',
+    'mount': 'waiting',
+    'install': 'waiting',
+    'start': 'waiting',
+    'ready': 'waiting',
+  })
```

Comment:
I think I still prefer a single status value to make things easy to track. Also showing the red cross on steps when user terminate the process feels a bit too much to me. I think it might be introducing a separate new flag to indicate `isInteractive` and showing something like "Interactive Mode, wait for server to start".